### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
[DEMO LINK](https://Zelinskyi-Serhii.github.io/react_goods-selector/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,85 +1,93 @@
...
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+type State = {
+  status: string;
+};
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:9]: from `status` I expect to see something like `completed, rejected` or `healthy, in_error` or even http responds status code

but I saw 'Jam' there... please use more clear variable names 🙂

File: src/App.tsx [hunk 2]
```diff
@@ -1,85 +1,93 @@
...
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {!status ? (<h1 className="title">No goods selected</h1>) : ''}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:29]: Why to return empty string from ternary operator? You can just use logical operator &&

File: src/App.tsx [hunk 3]
```diff
@@ -1,85 +1,93 @@
...
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => (
+                this.setState({ status: '' })
+              )}
+            />
+          </h1>
+        )
+          : ('')}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:46]: Even better, you can render one `h1` here and conditionally select attributes values here

File: src/App.tsx [hunk 4]
```diff
@@ -1,85 +1,93 @@
...
+                  'has-background-success-light': status === good,
+                })}
+              >
+                <td>
+                  {status === good ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => this.removeGood()}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:64]:
```suggestion
                      onClick={this.removeGood}
```

Comment (OleksiiBlezniuk) [reply src/App.tsx:64]: No need to create extra wrapper, just pass callback if you don't need to pass any arguments to it

File: src/App.tsx [hunk 5]
```diff
@@ -1,85 +1,93 @@
...
+                  ) : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => this.addGood(good)}
+                    >
+                      +
+                    </button>
+                  )}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:77]: Better to render one button and conditionally select needed values to pass

follow DRY

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -1,85 +1,91 @@
...
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.removeGood}
+            />
+          </h1>
+        )
+          : ('')}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:44]: Now you splited one check into two, either use ternary or logical operator &&

File: src/App.tsx [hunk 7]
```diff
@@ -1,85 +1,91 @@
...
+                <tr
+                  data-cy="Good"
+                  key={good}
+                  className={classNames({
+                    'has-background-success-light': selectedGood === good,
+                  })}
+                >
+                  <td>
+                    <button
+                      data-cy={isSelected ? ('RemoveButton') : ('AddButton')}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:61]:
```suggestion
                      data-cy={isSelected
                        ? 'RemoveButton'
                        : 'AddButton'}
```

File: src/App.tsx [hunk 8]
```diff
@@ -1,85 +1,91 @@
...
+                      type="button"
+                      className={classNames(
+                        'button',
+                        {
+                          'is-info': isSelected,
+                        },
+                      )}
+                      onClick={isSelected
+                        ? (this.removeGood)
+                        : (() => this.addGood(good))}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:71]: Nice, but would be better to save it to constant

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 9]
```diff
@@ -1,85 +1,95 @@
...
-];
+type State = {
+  selectedGood: string;
+};
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component<{}, State> {
+  state = {
```
Comment (sergii-nosachenko) [inline src/App.tsx:12]: Make it Readonly

File: src/App.tsx [hunk 10]
```diff
@@ -1,85 +1,95 @@
...
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {
+          selectedGood ? (
```
Comment (sergii-nosachenko) [inline src/App.tsx:30]:
```suggestion
        {selectedGood
          ? (
```

File: src/App.tsx [hunk 11]
```diff
@@ -1,85 +1,95 @@
...
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```
Comment (sergii-nosachenko) [inline src/App.tsx:34]: Don't disable linter, use `aria-label` instead

File: src/App.tsx [hunk 12]
```diff
@@ -1,85 +1,95 @@
...
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.removeGood}
+              />
+            </h1>
+          )
+            : (<h1 className="title">No goods selected</h1>)}
```
Comment (sergii-nosachenko) [inline src/App.tsx:43]:
```suggestion
            : (
              <h1 className="title">No goods selected</h1>
            )}
```

File: src/App.tsx [hunk 13]
```diff
@@ -1,85 +1,95 @@
...
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = selectedGood === good;
+              const addOrRemoveGood = isSelected
+                ? (this.removeGood)
+                : (() => this.addGood(good));
```
Comment (sergii-nosachenko) [inline src/App.tsx:51]:
```suggestion
                ? this.removeGood
                : () => this.addGood(good);
```

File: src/App.tsx [hunk 14]
```diff
@@ -1,85 +1,95 @@
...
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  key={good}
+                  className={classNames({
+                    'has-background-success-light': selectedGood === good,
```
Comment (sergii-nosachenko) [inline src/App.tsx:58]: Use `isSelected` here

File: src/App.tsx [hunk 15]
```diff
@@ -1,85 +1,95 @@
...
+                  })}
+                >
+                  <td>
+                    <button
+                      data-cy={isSelected
+                        ? ('RemoveButton')
+                        : ('AddButton')}
```
Comment (sergii-nosachenko) [inline src/App.tsx:65]: Unnecessary brackets

File: src/App.tsx [hunk 16]
```diff
@@ -1,85 +1,95 @@
...
+                        'button',
+                        {
+                          'is-info': isSelected,
+                        },
+                      )}
+                      onClick={addOrRemoveGood}
+                    >
+                      {isSelected
+                        ? ('-')
+                        : ('+')}
```
Comment (sergii-nosachenko) [inline src/App.tsx:77]: Unnecessary brackets

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Pull request closed without merging**